src/apps/workspaces/crud/user_applet_access.py

```diff
@@ -671,20 +720,67 @@ async def get_workspace_managers(
             .where(
                 UserAppletAccessSchema.soft_exists(),
                 UserAppletAccessSchema.owner_id == owner_id,
-                UserAppletAccessSchema.role != Role.RESPONDENT,
                 has_access,
+                UserAppletAccessSchema.role != Role.RESPONDENT,
                 UserAppletAccessSchema.applet_id == applet_id if applet_id else True,
             )
-            .group_by(UserSchema.id, UserAppletAccessSchema.title)
+            .group_by(UserSchema.id)
+        ).cte("accepted_users")
+
+        query: Query = (
+            select(
+                func.coalesce(accepted_users.c.id, invited_users.c.id).label("id"),
+                func.coalesce(accepted_users.c.first_name, invited_users.c.first_name).label("first_name"),
+                func.coalesce(accepted_users.c.last_name, invited_users.c.last_name).label("last_name"),
+                func.coalesce(accepted_users.c.email_encrypted, invited_users.c.email).label("email_encrypted"),
+                func.coalesce(accepted_users.c.created_at, invited_users.c.created_at).label("created_at"),
+                func.coalesce(accepted_users.c.last_seen, invited_users.c.created_at).label("last_seen"),
```

Comment:
Just to keep the same response and make it backward compatible

Reply:
The previous implementation was doing `func.coalesce(UserSchema.last_seen_at, UserSchema.created_at).label("last_seen")`, which you're doing here as part of the `accepted_users` query. Coalescing with `invited_users.c.created_at` is new behaviour.

I think we can still keep the `last_seen` property as-is and it will remain backwards compatible. Otherwise we will have a value show up in the last seen column for a pending invited user, which will be weird

Reply:
we have this first coalesce in accepted_users query